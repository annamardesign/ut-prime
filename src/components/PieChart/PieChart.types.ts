import { PaletteType } from 'devextreme/viz/palette';

export interface IPieChartProps {
    className?: string;
    dataSource: any[];
    title: string;
    palette?: PaletteType;
    argumentField: string;
    valueField: string;
}
