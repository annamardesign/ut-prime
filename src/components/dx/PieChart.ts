export { default } from 'devextreme-react/pie-chart';
