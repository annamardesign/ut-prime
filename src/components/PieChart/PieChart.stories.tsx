import React from 'react';
import { withReadme } from 'storybook-readme';

// @ts-ignore: md file and not a module
import README from './README.md';
import PieChart from './index';

export default {
    title: 'PieChart',
    component: PieChart,
    decorators: [withReadme(README)],
};

const data = [
    {
        country: 'Russia',
        area: 12,
    },
    {
        country: 'Canada',
        area: 7,
    },
    {
        country: 'USA',
        area: 7,
    },
    {
        country: 'China',
        area: 7,
    },
    {
        country: 'Brazil',
        area: 6,
    },
    {
        country: 'Australia',
        area: 5,
    },
    {
        country: 'India',
        area: 2,
    },
    {
        country: 'Others',
        area: 55,
    },
];

export const Basic: React.FC<{}> = () => (
    <PieChart dataSource={data} title="Area of Countries" argumentField="country" valueField="area" />
);
