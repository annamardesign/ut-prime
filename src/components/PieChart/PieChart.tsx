import React from 'react';
import classnames from 'classnames';
import DXPieChart, { Series, Label, Connector, Size, Export } from 'devextreme-react/pie-chart';

import { IPieChartProps } from './PieChart.types';
import './PieChart.css';

const PieChart: React.FC<IPieChartProps> = ({
    className,
    dataSource = [],
    title,
    palette = 'Bright',
    argumentField,
    valueField,
    ...rest
}) => {
    return (
        <div className={classnames('utfd-piechart', className)} {...rest}>
            <DXPieChart dataSource={dataSource} palette={palette} title={title}>
                <Series argumentField={argumentField} valueField={valueField}>
                    <Label visible={true}>
                        <Connector visible={true} width={1} />
                    </Label>
                </Series>

                <Size width={500} />
                <Export enabled={true} />
            </DXPieChart>
        </div>
    );
};

export default PieChart;
