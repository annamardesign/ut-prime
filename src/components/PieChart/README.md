# Button

## How to use

```jsx
import PieChart from 'ut-front-devextreme/core/PieChart';

<PieChart
    dataSource={[
        {
            country: 'USA',
            area: 7,
        }
    ]}
    title="Area of Countries"
    argumentField="country"
    valueField="area"
/>;
```

## Props

- **argumentField and valueField** - should be keys of an object from dataSource array

| propName      | propType           | defaultValue | isRequired |
| ---------     | --------           | ------------ | ---------- |
| dataSource    | array of objects   |              | yes        |
| title         | string             |              | yes        |
| argumentField | string             |              | yes        |
| valueField    | string             |              | yes        |
| palette       | PaletteType        | Bright       | yes        |

  