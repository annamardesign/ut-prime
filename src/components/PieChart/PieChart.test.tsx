import React from 'react';
import { render } from '@testing-library/react';

import PieChart from './index';

describe('<PieChart />', () => {
    it('renders component without breaking', () => {
        const { getByTestId } = render(
            <PieChart
                data-testid="PieChart"
                dataSource={[
                    {
                        country: 'Russia',
                        area: 12,
                    },
                ]}
                title="Area of Countries"
                argumentField="country"
                valueField="area"
            />,
        );

        expect(getByTestId('PieChart')).toMatchSnapshot();
    });
});
