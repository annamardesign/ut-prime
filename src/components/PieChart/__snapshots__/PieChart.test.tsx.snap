// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PieChart /> renders component without breaking 1`] = `
<div
  class="utfd-piechart"
  data-testid="PieChart"
>
  <div
    class="dx-visibility-change-handler"
  >
    <svg
      class="dxc dxc-chart"
      direction="ltr"
      fill="none"
      height="400"
      stroke="none"
      stroke-width="0"
      style="line-height:normal;-ms-user-select:none;-moz-user-select:none;-webkit-user-select:none;-webkit-tap-highlight-color:rgba(0, 0, 0, 0);display:block;overflow:hidden;"
      version="1.1"
      width="500"
      xmlns="http://www.w3.org/2000/svg"
    >
      <defs>
        <filter
          height="200%"
          id="DevExpress_1"
          transform="translate(0,0)"
          width="200%"
          x="-50%"
          y="-50%"
        >
          <fegaussianblur
            in="SourceGraphic"
            result="gaussianBlurResult"
            stdDeviation="3"
          />
          <feoffset
            dx="2"
            dy="6"
            in="gaussianBlurResult"
            result="offsetResult"
          />
          <feflood
            flood-color="#d3d3d3"
            flood-opacity="0.8"
            result="floodResult"
          />
          <fecomposite
            in="floodResult"
            in2="offsetResult"
            operator="in"
            result="compositeResult"
          />
          <fecomposite
            in="SourceGraphic"
            in2="compositeResult"
            operator="over"
          />
        </filter>
        <clippath
          id="DevExpress_2"
        >
          <rect
            height="400"
            transform="translate(0,0)"
            width="500"
            x="0"
            y="0"
          />
        </clippath>
        <clippath
          id="DevExpress_4"
        >
          <rect
            height="0"
            transform="translate(0,0)"
            width="445"
            x="0"
            y="0"
          />
        </clippath>
      </defs>
      <rect
        fill="gray"
        height="400"
        opacity="0.0001"
        transform="translate(0,0)"
        width="500"
        x="0"
        y="0"
      />
      <g
        class="dxc-background"
      />
      <g
        class="dxc-title"
        clip-path="url(#DevExpress_4)"
        transform="translate(240,0)"
      >
        <text
          style="white-space:pre;font-size:28px;font-family:'Segoe UI Light', 'Helvetica Neue Light', 'Segoe UI', 'Helvetica Neue', 'Trebuchet MS', Verdana, sans-serif;font-weight:200;fill:#232323;cursor:default;"
          text-anchor="middle"
          transform="translate(0,0)"
          x="0"
          y="0"
        >
          Area of Countries
        </text>
      </g>
      <g
        class="dxc-series-group"
      >
        <g
          class="dxc-series"
        >
          <g
            class="dxc-markers"
          >
            <path
              d="M 240.00000 225.00000 A 0.00000 0.00000 0 1 0 240.00000 225.00000 M 240.00000 225.00000 A 0.00000 0.00000 0 1 1 240.00000 225.00000 Z"
              fill="#70c92f"
              stroke="#ffffff"
              stroke-linejoin="round"
              stroke-width="0"
              transform="translate(0,0)"
            />
          </g>
        </g>
      </g>
      <g
        class="dxc-labels-group"
      >
        <g
          class="dxc-labels"
          opacity="0.001"
          transform="translate(0,0)"
        >
          <g
            visibility="visible"
          >
            <path
              d="M 66 225 L 66 225 L 36 225"
              stroke="#70c92f"
              stroke-width="1"
              transform="translate(0.5,0.5)"
            />
            <g
              transform="translate(28,225)"
            >
              <rect
                fill="#70c92f"
                height="8"
                isEmpty="true"
                stroke="none"
                stroke-width="0"
                transform="translate(0,0)"
                width="16"
                x="-8"
                y="-4"
              />
              <text
                style="white-space:pre;fill:#ffffff;font-family:'Segoe UI', 'Helvetica Neue', 'Trebuchet MS', Verdana, sans-serif;font-weight:400;font-size:12px;cursor:default;"
                transform="translate(0,0)"
                x="0"
                y="0"
              >
                12
              </text>
            </g>
          </g>
        </g>
      </g>
      <g
        class="dxc-legend"
        clip-path="url(#DevExpress_2)"
        transform="translate(0,0)"
      >
        <g
          transform="translate(490,61)"
        >
          <g
            class="dxc-item"
            style="fill:#767676;font-family:'Segoe UI', 'Helvetica Neue', 'Trebuchet MS', Verdana, sans-serif;font-weight:400;font-size:12px;cursor:default;"
          >
            <g>
              <g
                class="dxl-marker"
                transform="translate(0,0)"
                visibility="visible"
              >
                <rect
                  fill="#70c92f"
                  height="12"
                  opacity="1"
                  transform="translate(0,0)"
                  width="12"
                  x="0"
                  y="0"
                />
              </g>
              <text
                style="white-space:pre;"
                text-anchor="start"
                transform="translate(7,0)"
                x="0"
                y="0"
              >
                Russia
              </text>
            </g>
          </g>
        </g>
      </g>
      <g
        class="dxc-annotations"
      />
      <g
        class="dx-export-menu"
        hidden-for-export="true"
        transform="translate(455,8)"
      >
        <g
          class="dx-export-menu-button"
        >
          <rect
            cursor="pointer"
            fill="#ffffff"
            height="34"
            rx="4"
            ry="4"
            stroke="#ddd"
            stroke-width="1"
            transform="translate(0,0)"
            width="34"
            x="0.5"
            y="0.5"
          />
          <path
            cursor="pointer"
            d="M 9 12 L 26 12 L 26 14 L 9 14 M 9 17 L 26 17 L 26 19 L 9 19 M 9 22 L 26 22 L 26 24 L 9 24"
            fill="#333"
            transform="translate(0,0)"
          />
          <title>
            Exporting/Printing
          </title>
        </g>
      </g>
    </svg>
  </div>
</div>
`;
